2022/incoming/how-to-wrap-a-camel-.pod

```diff
@@ -0,0 +1,131 @@
+Author: Gavin Hayes <gahayes@cpan.org>
+Title: How to wrap a camel?
+Topic: Perl::Dist::APPerl
+
+=encoding utf8
+
+=head1 How to wrap a camel?
+
+In the elf workshop, elves never knew what to do when they were passed a camel and were instructed to wrap it to put it on Santa's sleigh. Many times the elves skipped wrapping and loaded camels right onto the sleigh. This practice garnered the North Pole many letters from children around the world being unable to install and use their new camel! B<"How do I install Perl?", "cpanm: command not found", "Permission denied", "collect2.exe: error: ld returned 1 exit status", "Can't locate"> they cried out for help. Some gift recipients even refused to install their camel despite their needs and desires, as they didn't want to deal with the difficulties of installing Perl. Claus tasked master elf, Artie, to design a fool-proof way of wrapping a camel for enjoyment by children across the world.
+
+Artie got right to work as they knew from experience that wrapping a camel was not a trivial task. First, Artie attempted to wrap a camel with L<PAR::Packer>, but discovered that the wrapped camel was unusable when the user's glibc version or operating system differed from the one Artie used to wrap. Artie knew Santa had high standards and would want a more robust solution, so they continued to do research. Artie stumbled upon the L<Cosmopolitan Libc|https://github.com/jart/cosmopolitan> and L<Actually Portable Executables|https://justine.lol/ape.html>. What if you built a single binary version of Perl that runs on six operating systems and melded your camel to the binary? Then, everyone can use camels! Artie got to work creating L<Actually Portable Perl (APPerl)|https://computoid.com/posts/Perl-is-Actually-Portable.html> with the L<Perl::Dist::APPerl> package for building APPerl and wrapping camels with it.
+
+Close to Christmas, Artie finally had L<Perl::Dist::APPerl> ready and notified Santa. Santa requested an immediate demonstration to determine if it is a viable way of wrapping camels starting this Christmas season.
+
+=head2 Wrapping Pure Perl camels
+
+Artie created some directories to contain the camel project.
+
+    mkdir -p wrap_camel/src wrap_camel/script
+    cd wrap_camel
+
+Artie created an example script and saved it to C<script/cameltype>:
+
+=begin perl
+
+    sub cancel_christmas { die "$!"; }
+    my $numhumps;
+    while(1) {
+        print "How many humps?\n";
+        $numhumps = <STDIN> or cancel_christmas();
+        chomp $numhumps;
+        last if($numhumps =~ /^\d+$/);
+        print "Your nose ain't too bright, $numhumps is not a number\n";
+    }
+    my %results = (1 => "Dromedary\n", 2 => "Bactrian\n");
+    print($results{$numhumps} // "Maybe a reindeer? Probably not a camel!\n");
+
+=end perl
+
+Artie noted to Santa that Pure Perl scripts can be wrapped with an existing version of APPerl, off the shelf builds of APPerl can be found at L<https://computoid.com/APPerl/#RELEASES> .
```

Comment:
I would give a text to this link

Reply:
done in https://github.com/perladvent/Perl-Advent/pull/239/commits/6c8f086089262f2995ac526f205bbdb5deea0f0d